### PR TITLE
Tickets/dm 41094

### PR DESCRIPTION
**Pull request by dsanmartim**
I'm opening this draft PR so you can help me with the script indexing. Tks!

### CODE REVIEW

File: python/lsst/ts/standardscripts/pause_queue.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+    """
+
+    def __init__(self, index):
+        super().__init__(index=index, descr="Pause the script queue.")
+
+        self.config = None
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/pause_queue.py:48]: I don't think you need to keep a copy of this.

File: python/lsst/ts/standardscripts/pause_queue.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration.
+        """
+        self.config = config
+
+        self.queue_index = getattr(SalIndex, self.config.queue)
+
+        # Initialize the script queue with the selected index
+        self.script_queue = ScriptQueue(
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/pause_queue.py:83]: I suggest you do:

```
if self.script_queue is None:
    self.script_queue = ScriptQueue(
        queue_index=self.queue_index, domain=self.domain, log=self.log
    )
```

File: python/lsst/ts/standardscripts/pause_queue.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    async def configure(self, config: types.SimpleNamespace) -> None:
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration.
+        """
+        self.config = config
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/pause_queue.py:78]: No need to keep this.

File: python/lsst/ts/standardscripts/pause_queue.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        )
+
+    def set_metadata(self, metadata):
+        pass
+
+    async def run(self):
+        """Run the script."""
+
+        self.log.info(
+            f"Pausing {self.config.queue} script queue. Resume the queue when you are ready..."
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/pause_queue.py:94]: Consider:

```
f"Pausing {self.queue_index!r} script queue. Resume the queue when you are ready..."
```

instead

File: tests/test_pause_queue.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+from lsst.ts.standardscripts import BaseScriptTestCase, get_scripts_dir
+from lsst.ts.standardscripts.pause_queue import PauseQueue
+
+
+class TestPauseQueue(BaseScriptTestCase, unittest.IsolatedAsyncioTestCase):
+    async def basic_make_script(self, index):
+        self.script = PauseQueue(index=index)
+
+        # Configure the script with a queue
```
Comment (tribeiro) [inline tests/test_pause_queue.py:34]: Remove the following 3 lines

File: tests/test_pause_queue.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        queue = "MAIN_TEL"
+        await self.configure_script(queue=queue)
+
+        # Mock the script queue and its start/pause method
+        self.script.script_queue = unittest.mock.MagicMock()
+        self.script.script_queue.start = unittest.mock.AsyncMock(return_value=None)
+        self.script.script_queue.pause = unittest.mock.AsyncMock(return_value=None)
+        self.script.script_queue.get_queue = unittest.mock.AsyncMock()
+
+        # Ensure the start method is called during script setup
```
Comment (tribeiro) [inline tests/test_pause_queue.py:44]: Remove these 2 lines as well.

**Review by tribeiro — COMMENTED** [review]
Here some initial comments.

File: tests/test_pause_queue.py (hunk 6 above)
Comment (dsanmartim) [reply tests/test_pause_queue.py:44]: The following two lines or lines 42 and 43?

Comment (tribeiro) [reply tests/test_pause_queue.py:44]: yes 42 and 43

Comment (dsanmartim) [reply tests/test_pause_queue.py:44]: I removed lines:

`# Ensure the start method is called during script setup
await self.script.script_queue.start()
`

**Review by tribeiro — APPROVED** [review]
Looks good!

**Pull request merged**